src/Nethermind/Nethermind.Serialization.Rlp/ReceiptMessageDecoder.cs

```diff
@@ -57,8 +57,14 @@ public TxReceipt Decode(RlpStream rlpStream, RlpBehaviors rlpBehaviors = RlpBeha
             {
                 entries[i] = Rlp.Decode<LogEntry>(rlpStream, RlpBehaviors.AllowExtraBytes);
             }
-
             txReceipt.Logs = entries;
+
+            if (rlpStream.Position != lastCheck)
```

Comment:
There is no nonce and version before canyon fork, so we need to check this way

Reply:
I mean don't try to decode nonce and version for non-deposits txs, it might cause us to try to decode malformed txs fields in mainnet for example. So checking the txType in addition to the current check would solve this.